portal/forms/home.py

```diff
@@ -47,5 +48,32 @@ class ContactForm(forms.Form):
     message = forms.CharField(label='Message', max_length=250,
                               widget=forms.Textarea(attrs={'class': 'contactField'}))
     browser = forms.CharField(label='Browser', max_length=250, required=False,
-                              widget=forms.TextInput(attrs={'type': 'hidden', 'id': 'browserField'})
-                              )
+                              widget=forms.TextInput(attrs={'type': 'hidden', 'id': 'browserField'}))
+    view_options = {'is_recaptcha_valid': False, 'is_recaptcha_visible': False}
+
+    def clean(self):
+        if self.view_options['is_recaptcha_visible']:
+            if not self.view_options['is_recaptcha_valid']:
```

Comment:
if self.view_options['is_recaptcha_visible'] and not self.view_options['is_recaptcha_valid']:   would be shorter and work the same (Python breaks if the left hand statement is False)